api/resource_groups.go

```diff
@@ -0,0 +1,207 @@
+//			ContainerTags:   []string{"*"},
+//		},
+//     },
+//   )
+//
+//   client.V2.ResourceGroups.Create(group)
+//
+func NewResourceGroup(name string, iType ResourceGroupType, props interface{}) ResourceGroupRaw {
+	return ResourceGroupRaw{
+		ResourceGroupData: ResourceGroupData{
+			Name:    name,
+			Type:    iType.String(),
+			Enabled: 1,
+			Props:   props,
+		},
+	}
+}
+
+// FindResourceGroupType looks up inside the list of available resource group types
+// the matching type from the provided string, if none, returns NoneResourceGroup
+func FindResourceGroupType(resourceGroup string) (ResourceGroupType, bool) {
+	for resType, resStr := range ResourceGroupTypes {
+		if resStr == resourceGroup {
+			return resType, true
+		}
+	}
+	return NoneResourceGroup, false
+}
+
+// List returns a list of Resource Groups
+func (svc *ResourceGroupsService) List() (response ResourceGroupsResponse, err error) {
+	err = svc.client.RequestDecoder("GET", apiV2ResourceGroups, nil, &response)
+	return
+}
+
+// Create creates a single Resource Group
+func (svc *ResourceGroupsService) Create(integration ResourceGroupRaw) (
+	response ResourceGroupsResponse,
+	err error,
+) {
+	err = svc.create(integration, &response)
+	return
+}
+
+// Delete deletes a Resource Group that matches the provided resource guid
+func (svc *ResourceGroupsService) Delete(guid string) error {
+	if guid == "" {
+		return errors.New("specify a resourceGuid")
+	}
+
+	return svc.client.RequestDecoder(
+		"DELETE",
+		fmt.Sprintf(apiV2ResourceGroupsFromGUID, guid),
+		nil,
+		nil,
+	)
+}
+
+// Get returns a raw response of the Resource Group with the matching resource guid.
+//
+// To return a more specific Go struct of a Resource Group, use the proper
+// method such as GetContainerResourceGroup() where the function name is composed by:
+//
+//  Get<Type>(guid)
+//
+//    Where <Type> is the Resource Group type.
+func (svc *ResourceGroupsService) Get(guid string, response interface{}) error {
+	return svc.get(guid, &response)
+}
+
+func (svc *ResourceGroupsService) create(data interface{}, response interface{}) error {
+	return svc.client.RequestEncoderDecoder("POST", apiV2ResourceGroups, data, response)
+}
+
+func (svc *ResourceGroupsService) get(guid string, response interface{}) error {
+	if guid == "" {
+		return errors.New("specify an resourceGuid")
+	}
+	apiPath := fmt.Sprintf(apiV2ResourceGroupsFromGUID, guid)
+	return svc.client.RequestDecoder("GET", apiPath, nil, response)
+}
+
+func (svc *ResourceGroupsService) update(guid string, data interface{}, response interface{}) error {
+	if guid == "" {
+		return errors.New("specify an intgGuid")
+	}
+	apiPath := fmt.Sprintf(apiV2ResourceGroupsFromGUID, guid)
+	return svc.client.RequestEncoderDecoder("PATCH", apiPath, data, response)
+}
+
+func (group ResourceGroupData) ResourceGroupType() ResourceGroupType {
+	t, _ := FindResourceGroupType(group.Type)
+	return t
+}
+
+type ResourceGroupRaw struct {
+	ResourceGroupData
+}
```

Comment:
lets stick with only one of `ResourceGroupRaw` or `ResourceGroupData`but we don't need both